torch/onnx/utils.py

```diff
@@ -686,7 +687,7 @@ def _optimize_graph(
     _C._jit_pass_lint(graph)
     graph = _C._jit_pass_canonicalize(graph)
     _C._jit_pass_lint(graph)
-    if GLOBALS.onnx_shape_inference:
+    if GLOBALS.onnx_shape_inference and not skip_shape_inference:
```

Comment:
I'm a little unsure about this. I recall the motivation is to correct shape/type after passes like `onnx_peephole`.

Reply:
OK, I've removed this change.